OpenRA.Mods.Common/Traits/Infantry/TakeCover.cs

```diff
@@ -38,22 +56,25 @@ public class TakeCover : Turreted, INotifyDamage, IDamageModifier, ISpeedModifie
 		public bool IsModifyingSequence { get { return IsProne; } }
 		public string SequencePrefix { get { return info.ProneSequencePrefix; } }
 
+		readonly HashSet<string> proneTriggers;
+
 		public TakeCover(ActorInitializer init, TakeCoverInfo info)
 			: base(init, info)
 		{
 			this.info = info;
+			proneTriggers = info.ProneTriggers.ToHashSet();
 		}
 
 		public void Damaged(Actor self, AttackInfo e)
 		{
-			/* Don't go prone when healed */
-			if (e.Damage > 0 && (e.Warhead == null || !e.Warhead.PreventProne))
-			{
-				if (!IsProne)
-					localOffset = info.ProneOffset;
-
-				remainingProneTime = info.ProneTime;
-			}
+			// Don't go prone when healed
+			if (e.Damage <= 0 || (e.Warhead != null && !e.Warhead.DamageTypes.Any(x => proneTriggers.Contains(x))))
```

Comment:
This will make infantry go prone if `e.Warhead == null`, which I think is the wrong thing to do.